domain/status/status.go

```diff
@@ -88,3 +96,6 @@ type UnitWorkloadStatuses map[coreunit.Name]UnitStatusInfo[WorkloadStatusType]
 // UnitCloudContainerStatuses represents the cloud container statuses of a collection
 // of units. The statuses are indexed by unit name.
 type UnitCloudContainerStatuses map[coreunit.Name]StatusInfo[CloudContainerStatusType]
+
+// FullUnitStatuses represents the workload and agent statuses of a collection of units.
+type FullUnitStatuses map[coreunit.Name]FullUnitStatus
```

Comment:
Why make this a map rather than a list of structs that include the unit name? You just loop through it anyway.